### PR TITLE
All active tests transformed from JUnit1-4 to Junit5

### PR DESCRIPTION
**Pull request by dmatej**
Changes were limited to src/test directories and test-utils module.
Main changes:

- hk2-runner was replaced by Hk2Junit5Extension and it's children
- as a consequence, utils module was rewritten from the ground and finally renamed
- many tests were reenabled and fixed
- remaining disabled tests detected bugs or are not stable or have some external dependencies

### Notes for reviewers

Please run the build including tests on your machine and report if it would fail - which module, which tests and don't forget to mention your system, JDK version and please, add stacktrace.

I am especially curious if there would not be any issues on Windows or Mac.

`mvn clean install -Pstaging -T4C -Pjacoco`

Tested on:
```
Kubuntu 20.10
Maven home: /home/dmatej/work/apache-maven-3.8.1
Java version: 17-ea, vendor: Oracle Corporation, runtime: /usr/lib/jvm/jdk-17-ea25-2522
Default locale: cs_CZ, platform encoding: UTF-8
OS name: "linux", version: "5.8.0-63-generic", arch: "amd64", family: "unix"
```

### CODE REVIEW

File: appserver/admin/admin-core/src/test/java/com/sun/enterprise/admin/tests/UpgradeTest.java [hunk 1]
```diff
@@ -16,63 +17,106 @@
...
+@ExtendWith(HK2JUnit5Extension.class)
+@DomainXml(value = "UpgradeTest.xml")
+public class UpgradeTest {
+
+    @Inject
+    private ServiceLocator locator;
+    @Inject
+    private StartupContext startupContext;
+
+    // FIXME: Workaround, because ServerEnvironmentImpl changes global System.properties, but other services
```
Comment (dmatej) [inline appserver/admin/admin-core/src/test/java/com/sun/enterprise/admin/tests/UpgradeTest.java:66]: Based on this I created this issue: https://github.com/eclipse-ee4j/glassfish/issues/23593

**Comment by pzygielo** [comment]
Passed with no problem on
```
Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)
Maven home: /usr/local/maven
Java version: 11.0.11, vendor: Oracle Corporation, runtime: /usr/local/jdk-11.0.11+9
Default locale: en_GB, platform encoding: UTF-8
OS name: "linux", version: "5.13.14-200.fc34.x86_64", arch: "amd64", family: "unix"
```
I've only checked that in code just tests were modified, and that junit4 was removed from nucleus/parent.

**Review by arjantijms — APPROVED** [review]
Amazing work! Just checked that it passed and glanced over the code. Looks good!

**Comment by arjantijms** [comment]
Forgot to post the versions:

```
Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)
Maven home: /usr/local/Cellar/maven/3.8.2/libexec
Java version: 17-ea, vendor: Oracle Corporation, runtime: /Users/arjan/.sdkman/candidates/java/17.ea.33-open
Default locale: en_NL, platform encoding: UTF-8
OS name: "mac os x", version: "11.5", arch: "x86_64", family: "Mac"
```

**Pull request merged**